article/_posts/2021-07-13-CoopTrainingAndLatentSpaceDA.md

```diff
@@ -0,0 +1,89 @@
+autoencoder, with everything being trained end-to-end.
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/figure1.jpg)
+
+## Latent space data augmentation
+The authors propose to randomly generate masks to apply to the latent space to drop features and generate corrupted, out
+of distribution data. In other words, they propose a form of dropout on the latent space vectors.
+
+Although not explained directly in the paper, the latent space learned seems to be structured such that
+$$z \in \mathbb{R}^{c \times h \times w}$$. The authors were thus able to use this structure to propose a few different
+schemes for generating the latent masks.
+
+1. **Random Masking with Dropout**: Drop random channels from the latent space.
+2. **Targeted Masking**: This scheme seeks to mask the most *salient* features, which can be determined by using two
+different methods. First, the gradients on the latent space have to be computed. Afterwards, either the i-th (i) pixels
+or (ii) channels with the highest mean gradients are masked. In this case, the features are not systemically dropped but
+rather they are scaled by an annealing factor $$a \in (0,0.5)$$.
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/figure3.jpg)
+
+## Cooperative training
+By cooperative training, the authors mean that their method is trained on both standard data and data corrupted
+on-the-fly. Thus, the dual-network is exposed to three hard example pairs, i.e. corrupted images-clean images
+($$\hat{x}$$, $$x$$), corrupted images-GT ($$\hat{x}$$, $$y$$), and corrupted prediction-GT ($$\hat{p}$$, $$y$$). The
+overall loss of the network thus becomes $$\mathcal{L}_{cooperative} = \mathcal{L}_{std} + \mathcal{L}_{hard}$$, where
+$$\mathcal{L}_{hard}$$ is defined as:
+
+$$
+\mathcal{L}_{hard} = \mathbb{E}_{\hat{x},\hat{p},x,y} [ \mathcal{L}_{rec}(\mathcal{D}_{\phi_i}(E_{\theta}(\hat{x})),x) + \mathcal{L}_{seg}(\bar{p},y) + \mathcal{L}_{shp}(\mathcal{C}_{\psi}(\hat{p}),y) + \mathcal{L}_{shp}(\mathcal{C}_{\psi}(\bar{p}),y) ]
+$$
+
+where
+
+$$
+\bar{p} = \mathcal{D}_{\phi_i} (\mathcal{H}(E_{\theta}(\hat{x}))), \quad \text{the FTN's predicted segmentation on}~\hat{x}.
+$$
+
+
+# Data
+To test their method, the authors based themselves on the ACDC[^2] and M&Ms[^3] dataset. The method was trained on 10
+patient randomly sampled from ACDC's training set, and the multi-site M&Ms dataset only served for cross-domain test.
+The authors also used another 20 patients from the ACDC dataset, which they augmented with different types of MR
+artifacts to generate challenging intra-domain subsets.
+
+
+# Results
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/table1.jpg)
+![](/article/images/CoopTrainingAndLatentSpaceDA/table2.jpg)
+
+
+# References
+- Code is available on GitHub: [https://github.com/cherise215/Cooperative_Training_and_Latent_Space_Data_Augmentation](https://github.com/cherise215/Cooperative_Training_and_Latent_Space_Data_Augmentation)
```

Comment:
Maybe they'll release it after the conference. For now, it contains no code.

Reply:
I saw that as well. In their paper, it is written code _will be_ available there. But I figured I didn't need to include this disclaimer, to avoid having to edit it eventually.

Reply:
OK. Maybe saying "Project code repository" is a good compromise.

Reply:
Done in fc4f62a.